### PR TITLE
added retry policy for Unauthorized error

### LINKED ISSUE #1508: Attempting to update booking status from iWTX returns an Unauthorized response
On https://manage.happytravel.com/

**Steps.**
1. Log in any admin;
2. Find the booking with Reference code HTL-MA-0008N2-01 https://manage.happytravel.com/booking/HTL-MA-0008N2-01 
3. Open the dev. tools in the browser;
4. Update the booking status;
5. Pay attention to the /refresh-status request.

**Result.**
Attempting to update booking status from iWTX returns an Unauthorized response.
https://edo.happytravel.com/en/api/1.0/admin/accommodations/bookings/808/refresh-status
```
{
    "status": 400,
    "detail": "Unauthorized"
}
```

**Expectations.**
The booking status should be updated without error.

### PR DESCRIPTION
**Pull request by DFSko**
resolves #1508

### CODE REVIEW

File: Api/Infrastructure/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -759,6 +758,29 @@ private static IAsyncPolicy<HttpResponseMessage> GetDefaultRetryPolicy()
...
 
 
+        private static IAsyncPolicy<HttpResponseMessage> GetConnectorRetryPolicy(IServiceProvider serviceProvider)
+        {
+            return Policy<HttpResponseMessage>
+                .Handle<HttpRequestException>()
+                .OrResult(msg => ConnectorRetryStatuses.Contains(msg.StatusCode))
+                .WaitAndRetryAsync(new[]
+                    {
+                        TimeSpan.FromMilliseconds(500),
```
Comment (kirillta) [inline Api/Infrastructure/ServiceCollectionExtensions.cs:768]: Maybe, one have to add a jitter here

**Review by kirillta — COMMENTED** [review]
How retry helps against unauthorized errors?

**Comment by DFSko** [comment]
> How retry helps against unauthorized errors?

Odaeara implements dynamic signature keys
Sometimes token is still valid but signature key already changed

File: Api/Infrastructure/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (DFSko) [reply Api/Infrastructure/ServiceCollectionExtensions.cs]: Added

**Review by kirillta — APPROVED** [review]

**Pull request merged**